src/explorers/glib_explorer.py

```diff
@@ -0,0 +1,87 @@
+"""An explorer that uses goal-literal babbling (GLIB)."""
+
+import logging
+from typing import Callable, List, Set
+
+from gym.spaces import Box
+
+from predicators.src import utils
+from predicators.src.explorers.bilevel_planning_explorer import \
+    BilevelPlanningExplorer
+from predicators.src.explorers.random_options_explorer import \
+    RandomOptionsExplorer
+from predicators.src.option_model import _OptionModelBase
+from predicators.src.settings import CFG
+from predicators.src.structs import NSRT, ExplorationStrategy, GroundAtom, \
+    ParameterizedOption, Predicate, Task, Type
+
+
+class GLIBExplorer(BilevelPlanningExplorer):
+    """GLIBExplorer implementation.
+
+    Sample a certain number of goals, plan to each of them, and execute
```

Comment:
this is not what the code is doing. what the code is doing is sample goals, score each of them, and then try planning starting from the highest-scoring goal, terminating at the first goal for which we find a plan.

the current comment makes it sound like we're planning to every goal